Engine/SerializerPlugins/ObjectSerializer.cs

```diff
@@ -238,12 +238,23 @@ public virtual bool TryDeserializeObject(XElement element, ITypeData t, Action<o
                                     Action<object> setValue = x =>
                                     {
                                         var current = property.GetValue(newobj);
-                                        property.SetValue(newobj, x);
-                                        if (false == Equals(current, x))
-                                        { // for some value-like type, it may be needed
-                                            // to set the parent object when a property is changed
-                                            // example: complex test plan parameters.
-                                            setter(newobj);
+                                        // Some plugins configure event handlers for ChildTestSteps in their constructor.
+                                        // If the list is already initialized, we should add the content of the newly deserialized list to the current list instead of overwriting it.
+                                        if (x is TestStepList newList && current is TestStepList currentList)
+                                        {
+                                            currentList.AddRange(newList);
```

Comment:
If the test step constructor adds a couple of steps, would this cause some issues?